code/modules/atmospherics/machinery/components/binary_devices/volume_pump.dm

```diff
@@ -59,45 +62,34 @@
 	icon_state = on && is_operational ? "volpump_on-[set_overlay_offset(piping_layer)]" : "volpump_off-[set_overlay_offset(piping_layer)]"
 
 /obj/machinery/atmospherics/components/binary/volume_pump/process_atmos()
+	last_power_draw = 0
+
 	if(!on || !is_operational)
 		return
 
 	var/datum/gas_mixture/air1 = airs[1]
 	var/datum/gas_mixture/air2 = airs[2]
 
-// Pump mechanism just won't do anything if the pressure is too high/too low unless you overclock it.
-
-	var/input_starting_pressure = air1.returnPressure()
+	//var/input_starting_pressure = air1.returnPressure()
 	var/output_starting_pressure = air2.returnPressure()
 
-	if((input_starting_pressure < 0.01) || ((output_starting_pressure > 9000))&&!overclocked)
-		return
-
-	if(overclocked && (output_starting_pressure-input_starting_pressure > 1000))//Overclocked pumps can only force gas a certain amount.
-		return
-
-
-	var/transfer_ratio = transfer_rate / air1.volume
-
-	var/datum/gas_mixture/removed = air1.removeRatio(transfer_ratio)
-
-	if(!removed.get_moles())
+//The pump will do nothing if the output is past the max pump pressure.
+	if(output_starting_pressure > MAX_PUMP_PRESSURE)
 		return
 
-	if(overclocked)//Some of the gas from the mixture leaks to the environment when overclocked
-		var/turf/open/T = loc
-		if(istype(T))
-			var/datum/gas_mixture/leaked = removed.removeRatio(VOLUME_PUMP_LEAK_AMOUNT)
-			T.assume_air(leaked)
+	if(air1.temperature > 0)
+		//copied from air injectors
+		var/transfer_moles = (air1.returnPressure() * transfer_rate) / (air1.temperature * R_IDEAL_GAS_EQUATION)
```

Comment:
```suggestion
	if(!(air1.temperature > 0))
		return
	//copied from air injectors
	var/transfer_moles = (air1.returnPressure() * transfer_rate) / (air1.temperature * R_IDEAL_GAS_EQUATION)
```
use this guard clause instead of putting all this intended